### ISSUE TITLE
Support notification type to differentiate Start upsertSearchAttr and Close workflow 

### ISSUE THREAD

**Issue by longquanzheng** [cadence-oss/cadence-notification#17]
This is more complicated than https://github.com/cadence-oss/cadence-notification/issues/16
We need to add a new field in Cadence server to indicate whether or not the message is from Start, close or Upsert search attributes the workflow:

https://github.com/cadence-oss/cadence-notification/blob/master/service/notification.go#L8

**Comment by zedongh** [comment]
RecordWorkflowExecutionStarted  and UpsertWorkflowExecution have same implementation body.

https://github.com/uber/cadence/blob/0557c2b503fb26362bb5066df1ddd2257fa60001/common/persistence/elasticsearch/esVisibilityStore.go#L90
https://github.com/uber/cadence/blob/0557c2b503fb26362bb5066df1ddd2257fa60001/common/persistence/elasticsearch/esVisibilityStore.go#L139

**Comment by longquanzheng** [comment]
@zedongh yes, that’s why we need to work on a PR in cadence server repo for this
